src/iris/ui/__init__.py

```diff
@@ -69,41 +72,44 @@ def get_local(req, path):
     return requests.get('http://127.0.0.1:16649/v0/%s' % path, cookies=req.cookies).json()
 
 
-def send_file(req, resp):
-    if not req.path.startswith('/static'):
-        raise HTTPNotFound()
-    filepath = os.path.join(ui_root, req.path[1:])
-    try:
-        resp.stream = open(filepath, 'rb')
-        resp.stream_len = os.path.getsize(filepath)
-    except IOError:
-        raise HTTPNotFound()
-
-
-def static_assets(req, resp):
-    if req.path.endswith('.js'):
-        resp.content_type = 'text/javascript'
-    elif req.path.endswith('.css'):
-        resp.content_type = 'text/css'
-    send_file(req, resp)
-
+# Credit to Werkzeug for implementation
+def secure_filename(filename):
+    for sep in os.path.sep, os.path.altsep:
+        if sep:
+            filename = filename.replace(sep, ' ')
+    filename = str(_filename_ascii_strip_re.sub('', '_'.join(
+        filename.split()))).strip('._')
+    return filename
 
-def static_imgs(req, resp):
-    if req.path.endswith('.png'):
-        resp.content_type = 'image/png'
-    elif req.path.endswith('.jpg'):
-        resp.content_type = 'image/jpg'
-    elif req.path.endswith('.svg'):
-        resp.content_type = 'image/svg+xml'
-    send_file(req, resp)
 
+class StaticResource(object):
+    allow_read_only = False
+    frontend_route = True
 
-def static_fonts(req, resp):
-    if req.path.endswith('.woff'):
-        resp.content_type = 'application/font-woff'
-    elif req.path.endswith('.ttf'):
-        resp.content_type = 'application/octet-stream'
-    send_file(req, resp)
+    def __init__(self, path):
+        self.path = path.lstrip('/')
+
+    def on_get(self, req, resp, filename):
+        if req.path.endswith('.js'):
```

Comment:
Instead of the giant if block, can you see if something like this works?

```python
suffix = os.path.splitext(req.path)[1]
mimes = {
	'.js': 'text/javascript',
	# all the rest
}
resp.content_type = mimes.get(suffix)
```